src/Http/Http/src/Features/FormFeature.cs

```diff
@@ -82,13 +81,12 @@ public bool HasFormContentType
         get
         {
             // Set directly
-            if (Form != null)
+            if (_form != null)
             {
                 return true;
             }
 
-            var contentType = ContentType;
-            return HasApplicationFormContentType(contentType) || HasMultipartFormContentType(contentType);
+            return HttpExtensions.HasApplicationFormContentType(_request.ContentType) || HttpExtensions.HasMultipartFormContentType(_request.ContentType);
```

Comment:
This is done a couple of times.

Add a `HasFormContentType(...)` to HttpExtensions that calls these methods for you? Eliminates chance of tests going out of sync.